pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -264,6 +261,126 @@ func (r *ReconcileMachineHealthCheck) Reconcile(ctx context.Context, request rec
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileMachineHealthCheck) remediate(ctx context.Context, currentHealthy []target, needRemediationTargets []target, errList []error, m *mapiv1.MachineHealthCheck) []error {
+	// remediate unhealthy
+	for _, t := range needRemediationTargets {
+		klog.V(3).Infof("Reconciling %s: meet unhealthy criteria, triggers remediation", t.string())
+		if m.Spec.RemediationTemplate != nil {
+			errList = r.externalRemediation(ctx, m, t, errList)
+		} else {
+			if err := t.internalRemediation(r); err != nil {
+				klog.Errorf("Reconciling %s: error remediating: %v", t.string(), err)
+				errList = append(errList, err)
+			}
+		}
+
+	}
+
+	for _, t := range currentHealthy {
+		if m.Spec.RemediationTemplate != nil {
```

Comment:
Personally I like to avoid nesting as much as possible. 
In this case for example, we can avoid nesting by using the other way around:
```golang
for _, t := range currentHealthy {
		if m.Spec.RemediationTemplate == nil {
                      continue
                }
                 // the nested code
}
```

something similar can be done in the previous for loop.
It's obviously subjective so feel free to keep it as you think fits.